editor/plugins/version_control_editor_plugin.h

```diff
@@ -49,46 +51,55 @@ class VersionControlEditorPlugin : public EditorPlugin {
 		CHANGE_TYPE_MODIFIED = 1,
 		CHANGE_TYPE_RENAMED = 2,
 		CHANGE_TYPE_DELETED = 3,
-		CHANGE_TYPE_TYPECHANGE = 4
+		CHANGE_TYPE_TYPECHANGE = 4,
+		CHANGE_TYPE_UNMERGED = 5
+	};
+
+	enum ButtonType {
+		BUTTON_TYPE_OPEN = 0,
+		BUTTON_TYPE_DISCARD = 1,
 	};
 
 private:
 	static VersionControlEditorPlugin *singleton;
 
-	int staged_files_count;
 	List<StringName> available_addons;
 
 	PopupMenu *version_control_actions;
 	AcceptDialog *set_up_dialog;
-	VBoxContainer *set_up_vbc;
-	HBoxContainer *set_up_hbc;
-	Label *set_up_vcs_label;
+	// VBoxContainer *set_up_vbc;
+	// HBoxContainer *set_up_hbc;
+	// Label *set_up_vcs_label;
 	OptionButton *set_up_choice;
 	PanelContainer *set_up_init_settings;
 	Button *set_up_init_button;
-	RichTextLabel *set_up_vcs_status;
-	Button *set_up_ok_button;
+	// RichTextLabel *set_up_vcs_status;
+	// Button *set_up_ok_button;
+	LineEdit *set_up_username;
+	LineEdit *set_up_password;
 
 	HashMap<ChangeType, String> change_type_to_strings;
 	HashMap<ChangeType, Color> change_type_to_color;
+	HashMap<ChangeType, Ref<Texture> > change_type_to_icon;
 
+	TabContainer *dock_vbc;
 	VBoxContainer *version_commit_dock;
-	VBoxContainer *commit_box_vbc;
-	HSplitContainer *stage_tools;
-	Tree *stage_files;
-	TreeItem *new_files;
-	TreeItem *modified_files;
-	TreeItem *renamed_files;
-	TreeItem *deleted_files;
-	TreeItem *typechange_files;
-	Label *staging_area_label;
-	HSplitContainer *stage_buttons;
-	Button *stage_all_button;
-	Button *stage_selected_button;
-	Button *refresh_button;
+	Tree *staged_files;
+	Tree *unstaged_files;
+	Tree *commit_list;
+
+	OptionButton *branch_select;
+
+	ToolButton *fetch_button;
+	ToolButton *pull_button;
+	ToolButton *push_button;
+
+	ToolButton *stage_all_button;
+	ToolButton *unstage_all_button;
+	ToolButton *discard_all_button;
+	ToolButton *refresh_button;
```

Comment:
What was the rationale behind converting all these to tool buttons?

Reply:
Tool buttons are used in other docks and match the editor theme. The normal button has a rectangle box with a border as a background that breaks the theme.